### PR TITLE
feat: notifications feed

### PR DESCRIPTION
**Pull request by fairlighteth**
1. The notifications bell and sidebar are displayed only when a wallet is connected
2. When there are no notifications, then "Nothing new yet" stub should be displayed
3. To get notifications list we poll https://cms.cow.fi/api/notification-list/0x{address} every 1 minute
4. All new notifications marked as read when user opens the notifications sidebar (`readNotificationsAtom:v0` in localStorage)

image

image

image

image

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/notifications/containers/ConnectTelegram.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+interface TelegramData {
+  auth_date: number
+  first_name: string
+  hash: string
+  id: number
+  photo_url: string
+  username: string
+}
+
+export function ConnectTelegram() {
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/notifications/containers/ConnectTelegram.tsx:17]: Copy-pasted from https://github.com/hprobotic/react-telegram-login/blob/master/src/index.js

**Comment by elena-zh** [comment]
Hey @fairlighteth , @shoom3301 
Here the list of issues I'd like to propose to fix: 
1. It would be great to have an exclamation sign or period in the end of the sentence (as we do everywhere)
image
2. It would be great to remove notifications from the widget (standalone mode) 
image
3. It would be great not to break words when wrapping to another line 
image
4. It would be great to limit the notification's displayed text with 3 lines, and then add 'View more' and 'View less' buttons to expand/collapse the notification's text 
image
5. Maybe not related to the current PR, but I'd like to propose to add a chain field into the notification's template. If a chain is specified, then displayed a notification in this chain only. If not, then to display on every chain. Why? IMO, this will be weird to see that I was sandwiched on Mainnet when I trade mostly on GC or so. 
6. Notification is labeled as unread each time when I refresh the page
7. Notification is labeled as unread each time I change account in a connected wallet (when to all users). IMO, the notification state should be stored in the local storage and should be labeled as unread if is has never been opened
8. It would be great to enhance a scrollbar in win11 
image
9. When I delete a notification, the red dot is still displayed in the app. When I open notifications bar, the deleted notification blinks and disappears there. IMO, it would be nice to hide the dot when a notification was unread and was removed from the cms. 
10. Something is wrong with the order of notifications displaying. See: dates are sorted asc, a new notification appears at the bottom. This happens when:
- several notifications were published to all users
- then I create a new notification to a certain user re-using template from the previous date
image
The order should be: 
 - dates: desc (16 of May, 15 of May)
 - Notifications publishing datetime within each date: newly published notification appears on top
12. (maybe not for this PR): not clear at all that in order to get a notification displayed in the app I need to select push = false. I'd replace this section with checkboxes section with multiple choices like: push/telegram/email/etc. 
image

Thanks!

**Review by anxolin — APPROVED** [review]
Maybe god to consolidate and we hide for now the telegram part

**Comment by shoom3301** [comment]
@elena-zh thank you!

1. Fixed
2. Removed it from Widget at all (standalone/dapp)
3. Fixed
4. I don't think we should do it now, currently the notifications are designed to be short
5. Out of the MVP scope
6. Please clean your `localStorage`. It's because of an outdated bug
7. Same as 6
8. Fixed
9. Out of the MVP scope
10. I wasn't able to reproduce. Please give me an account if it happens again
12. Out of the MVP scope

**Comment by elena-zh** [comment]
Hey @shoom3301 , thanks for fixes!
I've created #4443, #4444, #4445 and #4446 tasks for cases you labeled as not for MVP. 
As for the fixes: 
1. (maybe not related): dapp mode in the widget shows that I'm connected, but I'm not
dapp mode - conn to mm
2. I've cleared out my local storage, but cases 6-7 are reproducible to me (`Notification is labeled as unread each time when I refresh the page`,  `Notification is labeled as unread each time I change account in a connected wallet (when to all users). IMO, the notification state should be stored in the local storage and should be labeled as unread if is has never been opened`)
3. For the case 10, check this account `0x9FA3c00a92Ec5f96B1Ad2527ab41B3932EFEDa58`
4. I think, it would be great not to allow users modify due date in already published **AND outdated** notification template. Look: I've updated the date, and received 2 the same notifications (this may fix case 10?) 
update date - 2 notifications appeared

Thanks

**Comment by shoom3301** [comment]
@elena-zh thank you!

1. Not related to this PR
2. Checked on Zoom, localStorage clearing helps
3. Fixed
4. Unfortunately there is no such feature in Strapi, let's keep it as GTH in the feature

**Review by elena-zh — APPROVED** [review]
Additionally, I've created #4450 task to address case 4 from https://github.com/cowprotocol/cowswap/pull/4409#issuecomment-2117517510 in a separate task

**Pull request merged**